asset/asset.go

```diff
@@ -488,6 +488,85 @@ func (g *GroupKey) IsEqualGroup(otherGroupKey *GroupKey) bool {
 	return g.GroupPubKey.IsEqual(&otherGroupKey.GroupPubKey)
 }
 
+// IsGroupSig return true if the passed witness is a single Schnorr signature,
+// which is used to perform an asset group reissuance with a signature instead
+// of a script witness.
+func IsGroupSig(witness wire.TxWitness) *schnorr.Signature {
+	var potentialSig []byte
+	if len(witness) == 1 {
+		potentialSig = witness[0]
+	}
+
+	// We can safely drop the signature parsing error, since a witness that
+	// isn't a signature may still be a valid script witness.
+	if len(potentialSig) == schnorr.SignatureSize {
+		groupSig, _ := schnorr.ParseSignature(potentialSig)
+		return groupSig
+	}
+
+	return nil
```

Comment:
Maybe have a second boolean parameter instead of returning nil for an `Is...()` method?